wordpress-mobile/WordPress-Android
# Introduce updated locale picker

**_Note, this PR was originally created by @khaykov at https://github.com/wordpress-mobile/WordPress-Android/pull/14836. As the `develop` branch was deleted, it was not possible to reopen the original PR, which is why a new PR has been generated based on `trunk`._**

This PR updates the look of the app's locale picker, following the same design as the site timezone picker. It can be searched by both label and localized label.

[![Image from Gyazo](https://i.gyazo.com/ef74e9dab13e3dbd1fab40555d8e5b04.png)](https://gyazo.com/ef74e9dab13e3dbd1fab40555d8e5b04)

To test:
- Navigate to **App Settings**, and open the **Interface Language** option.
- Confirm that the new locale picker works as expected:
      - You can filter locales by both their label and localized label (eg. typing "Dani" and "Dans" should produce list with "Danish/Dansk" locale)
      - Tapping on locale changes app language.
      - Current app locale is displayed a the top of the list. Tapping on it closes the list.

## Regression Notes
1. Potential unintended areas of impact
- Locale Picker is not working!

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manual testing + VM tests.

3. What automated tests I added (or what prevented me from doing so)
- Hopefully all of them!

PR submission checklist:
